CMakeLists.txt

```diff
@@ -50,9 +50,9 @@ zephyr_library_named(zephyr)
 zephyr_include_directories(
   kernel/include
   arch/${ARCH}/include
-  soc/${ARCH}/${SOC_PATH}
-  soc/${ARCH}/${SOC_PATH}/include
-  soc/${ARCH}/${SOC_FAMILY}/include
+  ${SOC_DIR}/${ARCH}/${SOC_PATH}
```

Comment:
I wonder if we need a `SOC_DIR`. We already have a `BOARD_DIR`, so why not have instead a `ZEPHYR_OOT_DIR` that points to a folder that inside has `soc/` and `boards/`?

Reply:
@carlescufi The SOC and the board may be in separate git repositories and placed in separate paths.

Reply:
I know, but is it really important that we allow this? Wouldn't it be better to allow out of tree boards and SoCs, but keeping the same folder structure as in the Zephyr repo itself? Otherwise we might end up with a whole bunch of env vars like those?

Reply:
I guess it is not too important. It just adds a bit more of flexibility/convenience, but nothing one cannot work relatively easy around.

> Otherwise we might end up with a whole bunch of env vars like those?
An ARCH_DIR would be the next logical one I'd be happy to see :)

Reply:
> I wonder if we need a SOC_DIR. We already have a BOARD_DIR, so why not have instead a ZEPHYR_OOT_DIR that points to a folder that inside has soc/ and boards/?

the use case I am implementing right now is soc and board codes in different git repos. This was also tested with both in the same structure resembling the zephyr tree. (both variables point to the same thing)

I do not think having another variable is excessive here (compare with the 2839485 Kconfig variables we have :-)

Reply:
Fine, if you rely on that then no objections.